src/main/kotlin/me/owdding/catharsis/features/timespan/Timespans.kt

```diff
@@ -0,0 +1,143 @@
+package me.owdding.catharsis.features.timespan
+
+import com.google.gson.GsonBuilder
+import com.google.gson.JsonElement
+import com.mojang.serialization.Codec
+import me.owdding.catharsis.Catharsis
+import me.owdding.catharsis.events.FinishRepoLoadEvent
+import me.owdding.catharsis.events.StartRepoLoadEvent
+import me.owdding.catharsis.repo.CatharsisRemoteRepo
+import me.owdding.catharsis.utils.CatharsisLogger
+import me.owdding.catharsis.utils.CatharsisLogger.Companion.featureLogger
+import me.owdding.catharsis.utils.extensions.sendWithPrefix
+import me.owdding.catharsis.utils.types.colors.CatppuccinColors
+import me.owdding.ktmodules.Module
+import net.minecraft.resources.FileToIdConverter
+import net.minecraft.resources.Identifier
+import net.minecraft.server.packs.resources.ResourceManager
+import net.minecraft.server.packs.resources.SimplePreparableReloadListener
+import net.minecraft.util.profiling.ProfilerFiller
+import tech.thatgravyboat.skyblockapi.api.events.base.Subscription
+import tech.thatgravyboat.skyblockapi.api.events.base.predicates.TimePassed
+import tech.thatgravyboat.skyblockapi.api.events.misc.RegisterCommandsEvent
+import tech.thatgravyboat.skyblockapi.api.events.time.TickEvent
+import tech.thatgravyboat.skyblockapi.helpers.McClient
+import tech.thatgravyboat.skyblockapi.helpers.McLevel
+import tech.thatgravyboat.skyblockapi.utils.json.Json.toDataOrThrow
+import tech.thatgravyboat.skyblockapi.utils.text.Text
+import tech.thatgravyboat.skyblockapi.utils.text.Text.send
+import tech.thatgravyboat.skyblockapi.utils.text.TextBuilder.append
+import tech.thatgravyboat.skyblockapi.utils.text.TextStyle.color
+import java.io.Reader
+import kotlin.io.path.reader
+
+@Module
+object Timespans : SimplePreparableReloadListener<List<Pair<Identifier, TimespanDefinition>>>(), CatharsisLogger by Catharsis.featureLogger() {
```

Comment:
reorder the functions to make more sense the apply is below tick but the parsing is above the repo loading and the command registration is at the top